installscripts/Installer.sh

```diff
@@ -133,24 +153,24 @@ function install_packages () {
 
   #Download and install pip
   if command -v pip > /dev/null; then
-     spin_wheel $! "System-level pip install found, using that."
+      print_info "pip already installed, using it"
   else
      curl -sL $PIP_URL -o get-pip.py
      sudo python get-pip.py >>$LOG_FILE &
      spin_wheel $! "Downloading and installing pip"
   fi
 
-  # Download and Install awscli
-  sudo pip install awscli >> $LOG_FILE &
-  spin_wheel $! "Downloading & installing awscli bundle"
-
-  #Download and install paramiko
-  sudo pip install paramiko >>$LOG_FILE &
-  spin_wheel $! "Downloading and installing paramiko"
+  if command -v aws > /dev/null; then
+      print_info "awscli already installed, using it"
+  else
+      # Download and Install awscli
+      sudo pip install awscli >> $LOG_FILE &
+      spin_wheel $! "Downloading & installing awscli bundle"
+  fi
 
-  #Undo output redirection
-  exec 1> /dev/stdout
-  exec 2> /dev/stderr
+  #Undo output redirection and close unused file descriptors.
+  exec 1>&3 3>&-
```

Comment:
Thanks for fixing this, we were seeing symptoms of this in our daily stack where the jenkins box after a few builds will talk about one too many open file handles. We never got to understand the issue, and this makes me think its related.

Reply:
Ah, yes, it might be.